### PR TITLE
chore(rln-relay): refactor mounting procedure

### PR DESCRIPTION
**Pull request by rymnc**
Refactors the mounting procedure of waku-rln-relay

- [x] Updates the barrel to include the contract
- [x] Remove cyclic dependency on waku_node
- [x] Fix tests

### CODE REVIEW

**Comment by rymnc** [comment]
Checks are intermittently failing due to an unrelated error -
```
2022-12-09T10:02:24.8182700Z     /Users/runner/work/nwaku/nwaku/tests/v2/test_waku_filter.nim(172, 18): Check failed: await pushHandlerFuture.withTimeout(1.seconds)
```

**Comment by LNSD** [comment]
> Checks are intermittently failing due to an unrelated error -
> 
> ```
> 2022-12-09T10:02:24.8182700Z     /Users/runner/work/nwaku/nwaku/tests/v2/test_waku_filter.nim(172, 18): Check failed: await pushHandlerFuture.withTimeout(1.seconds)
> ```

Could you file this failure in #1357?

**Review by LNSD — APPROVED** [review]
LGTM ✅

A comment for future refactoring work: 

> If I understand the RLN module correctly, we have two mount protocol modes selected by the `rlnRelayDynamic` configuration flag. 
> 
> To boost this module's extensibility and maintainability (e.g. testability), we should generalize the RLN protocol by extracting an interface that covers both modes and provides this behaviour through the constructor (check the driver interface and how we "inject" it into the Waku archive instance).
>
> We can have a synchronous analysis and design session if needed 😁

**Comment by rymnc** [comment]
> 

Agreed, I'll do that as part of https://github.com/waku-org/nwaku/issues/1370, thanks for your review :)

File: tests/v2/test_wakunode_rln_relay.nim [hunk 1]
```diff
@@ -151,56 +131,41 @@ procSuite "WakuNode - RLN relay":
       rlnRelayPubSubTopic = RlnRelayPubsubTopic
       contentTopic = ContentTopic("/waku/2/default-content/proto")
 
+    # set up three nodes
+    # node1
```
[the remark this answers is not in the document]
Comment (rymnc) [reply tests/v2/test_wakunode_rln_relay.nim:135]: No, it isn't, if you're referring to different tests that setup 3 nodes, they test something different each time ;)

File: waku/v2/protocol/waku_rln_relay/utils.nim [hunk 2]
```diff
@@ -1309,59 +1297,81 @@ proc mount(node: WakuNode,
...
+                                               pubsubTopic = conf.rlnRelayPubsubTopic,
+                                               contentTopic = conf.rlnRelayContentTopic, 
+                                               spamHandler = spamHandler, 
+                                               registrationHandler = registrationHandler)
+
+    if rlnRelayRes.isErr():
+      return err("dynamic rln-relay could not be mounted: " & rlnRelayRes.error())
+    let wakuRlnRelay = rlnRelayRes.get()
+    if persistCredentials:
+      # TODO should be replaced with key-store with proper encryption
```
[the remark this answers is not in the document]
Comment (rymnc) [reply waku/v2/protocol/waku_rln_relay/utils.nim]: Indeed! addressed in [e8f4aaa](https://github.com/waku-org/nwaku/pull/1457/commits/e8f4aaada857e06033ff1e9990f1dc940e30c068)

**Pull request merged**